challenge_2/go/erocs/find_singleton.go

```diff
@@ -0,0 +1,29 @@
+package main
+
+import (
+	"fmt"
+)
+
+func FindSingleton(int_array []int) (found int) {
+	singles := make(map[int]int)
+	repeats := make(map[int]int)
+	for _, n := range int_array {
+		if _, ok := repeats[n]; ok {
+			repeats[n]++
+		} else if _, ok := singles[n]; ok {
+			delete(singles, n)
+			repeats[n] = 2
+		} else {
+			singles[n] = 1
+		}
+	}
+	for k, _ := range singles {
+		found = k
```

Comment:
If I had the return within the loop, I would still need the end the function return, since the loop might contain nothing. It doesn't gain anything since this loop should only iterate one value, per spec. If I updated to have both return statements, I would also change the return value to (int, error) and rework the loop so an error could be returned on no single or multiple singles. Not adding the error return was just my laziness.  ;)